compiler/optimizer/LoopVersioner.cpp

```diff
@@ -4314,6 +4314,49 @@ void TR_LoopVersioner::versionNaturalLoop(TR_RegionStructure *whileLoop, List<TR
       dumpOptDetails(comp(), "The node %p has been created for testing if virtual guard is required\n", nextComparisonNode);
       }
 
+   bool safeToRemoveOSRGuards = false;
```

Comment:
This could use a comment to describe the logic of when it is safe to remove an OSRGuard and there is an assumption in this code that all OSRGuards are the same which is currently true, but which could potentially have to change in the future if OSRGuards patch more specifically in the future.